pkg/resourcemanager/webhook/add.go

```diff
@@ -31,10 +33,26 @@ import (
 
 // AddToManager adds all webhook handlers to the given manager.
 func AddToManager(mgr manager.Manager, _, targetCluster cluster.Cluster, cfg *config.ResourceManagerConfiguration) error {
+	kubernetesClient, err := kubernetesclientset.NewForConfig(targetCluster.GetConfig())
+	if err != nil {
+		return fmt.Errorf("failed creating Kubernetes client: %w", err)
+	}
+
+	targetServerVersion, err := kubernetesClient.DiscoveryClient.ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	targetVersion, err := semver.NewVersion(targetServerVersion.GitVersion)
```

Comment:
This will fail if we have non-semver compliant versions, e.g. in GKE - [example](https://github.com/gardener/gardener/pull/6468).
We could pass the `targetSeverVersion` and use [version.CheckVersionMeetsConstraint](https://github.com/gardener/gardener/blob/a8e71c8c1b14c35bbe669ffceb6086daad527ada/pkg/utils/version/version.go#L140) in the handler.